### ISSUE TITLE
Make documentation more clear about using with React

### ISSUE THREAD

**Issue by JGallardo** [FortAwesome/react-fontawesome#293]
**Is your feature request related to a problem? Please describe.**
The problem is that the documentation misses gaps in information.

Like using icons with two words.

**Describe the solution you'd like**
Just add an actual example or screenshot of a real use case.

**Describe alternatives you've considered**

**Additional context**
All your documenation says is to add (as an example) `<FontAwesomeIcon icon="coffee" />`

But I found in a thread on line to do something like 
`<FontAwesomeIcon icon={["fab", "github"]} />`, which works fine, i did not even have to "import them to a library" like your documentation says to do. I can perfectly use SOME icons like `<FontAwesomeIcon icon={["fab", "acorn"]} />`

But what about when I want to use something like Amazon pay? well when i looked it up at 
[https://fontawesome.com/icons]( https://fontawesome.com/icons?d=gallery&m=free) I found it to say `amazon-pay`, which i used and it works fine.

But when i searched for something very common like an email icon, i found the envelope. So i tried `<FontAwesomeIcon icon={["fab", "envelope"]} />`
That does not work though. I did a quick google search and found a well documented example at https://alligator.io/react/font-awesome/
But what about when we need to do something like adding an envelope-open? after experimenting I got it working with `<FontAwesomeIcon icon="envelope-open"/>`

and having made the import with 
```
import { faEnvelope, faEnvelopeOpen } from '@fortawesome/free-solid-svg-icons';

library.add(fab, faEnvelope, faEnvelopeOpen)
```

But where would anyone learn to do that? that for example "envelope-open" should be written as "faEnvelopeOpen"?

**Comment by vickOnRails** [comment]
I second this.

The docs for React miss a lot of information.
I'm about an hour into the docs and can't figure out how to import a single icon.

😟

**Comment by jstafford74** [comment]
Just want to let you know that this is working for me.  I agree documentation could be better.  Was about 40min of brain damage....

//========./utils/initFontAwesome.js============//
```
import { library } from "@fortawesome/fontawesome-svg-core";
import {fab, faFacebook, faLinkedin, faGithub} from "@fortawesome/free-brands-svg-icons";

function initFontAwesome() {
    library.add(fab, faFacebook, faLinkedin, faGithub);
}
export default initFontAwesome;
```
//=========Component.jsx==========//
```
import React from 'react';
import Home from '../src/pages/home';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

<Nav.Link href="https://www.facebook.com/jeremiah.stafford.1">
      <FontAwesomeIcon icon={["fab", "facebook"]} className="mr-2" />
 </Nav.Link>
 <Nav.Link href="https://www.linkedin.com/in/jeremiah-stafford-frm-25a25667/">
       <FontAwesomeIcon icon={["fab", "linkedin"]} className="mr-2" />
  </Nav.Link>
  <Nav.Link href="https://github.com/jstafford74">
       <FontAwesomeIcon icon={["fab", "github"]} className="mr-2" />
  </Nav.Link>
```
//==========App.js===============//

```
import initFontAwesome from "./utils/initFontAwesome";
initFontAwesome();

function App() {
  return (
    <div className="App">
      <Home />
    </div>
  );
}
export default App;
```

**Comment by danilockthar** [comment]
I agree. I never have problems with other libraries like im having with font awesome react. So messed up

**Comment by mikejact** [comment]
I completely agree. Their documentation is not very clear. I imported LinkedIn no problem, but after searching forever I couldn't figure out how to import envelope until I stumbled across this article. Thanks for posting this!

**Comment by SevlaMare** [comment]
We could have an example of stacking icons on readme, like we do without components:

```
<span class="fa-stack fa-2x">
  <i class="fas fa-square fa-stack-2x"></i>
  <i class="fab fa-twitter fa-stack-1x fa-inverse"></i>
</span>
```

Actually could be all example from official page:
https://fontawesome.com/v4.7.0/examples/
That will be great, it already solve a lot of use cases.

**Comment by SevlaMare** [comment]
We could have an example of stacking icons on readme, like we do without components:

```
<span class="fa-stack fa-2x">
  <i class="fas fa-square fa-stack-2x"></i>
  <i class="fab fa-twitter fa-stack-1x fa-inverse"></i>
</span>
```

Actually could be all examples from official page:
https://fontawesome.com/v4.7.0/examples/
That will be great, it already solve a lot of use cases.

**Comment by ivan-kleshnin** [comment]
Yep, I've spent an hour trying to make stacking work with no luck.

**Comment by charles4221** [comment]
The docs have been updated significantly over the years so I will be closing this issue. If you believe there is still missing information or it is unclear how to use the features, please provide further detail based on the current iteration of the docs at https://docs.fontawesome.com
